Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,638 @@
+/**
+ * @fileoverview
+ * This file contains the majority of the functionality used across the various web pages, in the following order
+ * 1: universal functions
+ * 2: functions related to invoking methods
+ * 3: camera related functions
+ */
+
+
+//immutable global variables are written in all caps
+const PORT = 5000;
+
+/**
+ * @global variables. Mutable global variables in javascript are typically named in camelCase
+ */
+var graphTopologies={};
+var graphInstances={};
+var cameras={};
+
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// universal functions
+
+/**
+* get global variables from server, Returns a promise so that functions that depend on these variables can wait
+* @returns {Promise<any>}
+*/
+function getGlobals()
+{
+  return new Promise((resolve, reject) => 
+  {
+    var request = sendRequest("", `http://localhost:${PORT}/getglobals`, "GET");
+    request.onreadystatechange = function () 
+    {
+      if(request.readyState == 4 && request.status == 200)
+      {
+        let response= JSON.parse(request.response);
+        console.log(response);
+        for (var list in response)
+        {
+          let temp=response[list].name;
+          switch(temp)
+          {
+            case 'graphInstances':
+              graphInstances=response[list].value;
+              break;
+            case 'graphTopologies':
+              graphTopologies=response[list].value;
+              break;
+            case 'cameras':
+              cameras=response[list].value;
+              break;
+          }
+        }
+        resolve(response);
+      }
```

Comment:
if something fails (ex ready state never hits 4, status isn't 200) this seems broken... what's the expectation?

Reply:
Well I don't think this would ever run into the issue of state not reaching 4... Added a handler for if status isn't 200

Reply:
ready state 1 is "server connection established". if you fail to establish a server connection or lose it between stages 1 and 4 what happens?

Reply:
added timeout handler for all requests